bundle/src/main/java/com/cognifide/cq/cqsm/core/models/RowModel.java

```diff
@@ -22,83 +22,85 @@
 import com.cognifide.cq.cqsm.core.scripts.ScriptImpl;
 import com.day.cq.commons.jcr.JcrConstants;
 import com.google.common.collect.ImmutableSet;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Set;
+import javax.annotation.PostConstruct;
 import org.apache.commons.lang.StringUtils;
 import org.apache.sling.api.SlingHttpServletRequest;
 import org.apache.sling.api.resource.Resource;
 import org.apache.sling.models.annotations.Model;
 import org.apache.sling.models.annotations.injectorspecific.Self;
 
-import javax.annotation.PostConstruct;
-import java.util.Calendar;
-import java.util.Date;
-import java.util.GregorianCalendar;
-import java.util.Set;
-
 @Model(adaptables = SlingHttpServletRequest.class)
 public class RowModel {
 
-    @Self
-    private SlingHttpServletRequest request;
+  @Self
+  private SlingHttpServletRequest request;
 
-    private ScriptImpl script;
+  private ScriptImpl script;
 
-    private String scriptName;
+  private String scriptName;
 
-    private boolean isFolder;
+  private boolean isFolder;
 
-    private static final Set<String> FOLDER_TYPES = ImmutableSet.of(JcrConstants.NT_FOLDER, "sling:OrderedFolder");
+  private static final Set<String> FOLDER_TYPES = ImmutableSet
+      .of(JcrConstants.NT_FOLDER, "sling:OrderedFolder");
 
-    @PostConstruct
-    public void init() {
-        Resource resource = request.getResource();
-        script = resource.adaptTo(ScriptImpl.class);
-        scriptName = resource.getName();
-        isFolder = FOLDER_TYPES.contains(resource.getValueMap().get(JcrConstants.JCR_PRIMARYTYPE, StringUtils.EMPTY));
+  @PostConstruct
+  public void init() {
+    Resource resource = request.getResource();
+    isFolder = FOLDER_TYPES
+        .contains(resource.getValueMap().get(JcrConstants.JCR_PRIMARYTYPE, StringUtils.EMPTY));
+    scriptName = resource.getName();
+    if (!isFolder) {
+      script = resource.adaptTo(ScriptImpl.class);
     }
+  }
 
-    public String getScriptName() {
-        return scriptName;
-    }
+  public String getScriptName() {
+    return scriptName;
+  }
 
-    public String getAuthor() {
-        return script != null ? script.getAuthor() : "";
-    }
+  public String getAuthor() {
+    return script != null ? script.getAuthor() : "";

```

Comment:
You could use Optional API: `Optional.ofNullable(script).map(Script::getAuthor).orElse("")` - you could also change  `private ScriptImpl script` to ` private Optional<ScriptImpl> script;` which will simplify this.

Reply:
Done, refactored whole RowModel class.